ddht/v5_1/rpc_handlers.py

```diff
@@ -51,48 +80,47 @@ def validate_endpoint(value: Any) -> None:
         raise RPCError(f"Invalid Endpoint: {value}")
 
 
-class PingHandler(RPCHandler[Tuple[NodeID, Optional[Endpoint]], PongResponse]):
-    def __init__(self, network: NetworkAPI) -> None:
-        self._network = network
+def validate_and_extract_destination(value: Any) -> Tuple[NodeID, Optional[Endpoint]]:
+    node_id: NodeID
+    endpoint: Optional[Endpoint]
 
-    def extract_params(self, request: RPCRequest) -> Tuple[NodeID, Optional[Endpoint]]:
-        try:
-            raw_params = request["params"]
-        except KeyError as err:
-            raise RPCError(f"Missiing call params: {err}")
+    if is_hex_node_id(value):
+        node_id = NodeID(decode_hex(value))
+        endpoint = None
+    elif value.startswith("enode://"):
+        raw_node_id, _, raw_endpoint = value[8:].partition("@")
 
-        if len(raw_params) != 1:
-            raise RPCError(
-                f"`ddht_ping` endpoint expects a single parameter: "
-                f"Got {len(raw_params)} params: {raw_params}"
-            )
+        validate_hex_node_id(raw_node_id)
+        validate_endpoint(raw_endpoint)
+
+        node_id = NodeID(decode_hex(raw_node_id))
 
-        value = raw_params[0]
+        raw_ip_address, _, raw_port = raw_endpoint.partition(":")
+        ip_address = ipaddress.ip_address(raw_ip_address)
+        port = int(raw_port)
+        endpoint = Endpoint(ip_address.packed, port)
+    elif value.startswith("enr:"):
+        enr = ENR.from_repr(value)
+        node_id = enr.node_id
+        endpoint = Endpoint.from_enr(enr)
```

Comment:
If this raises an exception because the provided `enr` doesn't have enough information in it will that be convered into an `RPCError`?

Reply:
Good catch.  I'll add a test for this.